### ISSUE TITLE
What is "target_index" on BertForMaskedLM?

### ISSUE THREAD

**Issue by lucadinidue** [hmohebbi/ValueZeroing#1]
First of all, thanks for sharing the code! I'm trying to use ValueZeroing on some experiments, and I noticed that the class `BertForMaskedLM` has the parameter `target_index` initialized as None and used in this line:

```
# Added by Hosein  
prediction_scores = prediction_scores[torch.arange(prediction_scores.size(0)), target_index]
```

But in the demo it is not passed to the model during the forward pass, so it remains None (changing the dimensions of `prediciton_scores`?). So, what is this parameter used for? What am I missing?

Thanks a lot in advance,
Luca

**Comment by hmohebbi** [comment]
Hi Luca, thanks for your question! By setting the input parameter 'target_index' the model returns the logits predicted at a specific time step (in this case, the masked token). This simplifies the computation of logit-based scores (such as Gradient-based scores). The parameter is not used in the Demo file where Value Zeroing (VZ) is being computed, as VZ relies on the model's representations and is unrelated to the output logits.

Best of luck!
